hkmc2/shared/src/main/scala/hkmc2/semantics/Elaborator.scala

```diff
@@ -917,6 +938,13 @@ extends Importer:
           raise(ErrorReport(msg"Unrecognized definitional assignment left-hand side: ${lhs.describe}"
             -> lhs.toLoc :: Nil)) // TODO BE
           go(sts, Nil, Term.Error :: acc)
+      // A use[T] construct: analogous to Scala's summon[T].
+      case (tree @ TermDef(Ins, _, N)) :: sts =>
+        val res = annotations.foldLeft(term(tree)):
+          case (acc, ann) => Term.Annotated(ann, acc)
+        sts match
+        case Nil => (mkBlk(acc, S(res), hasResult), ctx)
+        case _ => go(sts, Nil, res :: acc)
```

Comment:
Here's a duplication of the fallback case that handles the trees (as opposed to statements)

```scala
      case (st: Tree) :: sts =>
        // TODO reject plain term statements? Currently, `(1, 2)` is allowed to elaborate (tho it should be rejected in type checking later)
        val res = annotations.foldLeft(term(st)):
          case (acc, ann) => Term.Annotated(ann, acc)
        sts match
        case Nil => (mkBlk(acc, S(res), hasResult), ctx)
        case _ => go(sts, Nil, res :: acc)
```

I think it is not beautiful but seems there is no other way to work around it.